File: sample-data-repository/src/main/java/nl/haploid/octowight/sample/data/CaptainModelSerializer.java
package nl.haploid.octowight.sample.data;

import nl.haploid.octowight.JsonMapper;
import nl.haploid.octowight.registry.data.ModelSerializer;
import org.springframework.stereotype.Component;

@Component
public class CaptainModelSerializer implements ModelSerializer<CaptainModel> {

	@Override
	public String serialize(final CaptainModel model) {
		final JsonMapper jsonMapper = new JsonMapper();
		return jsonMapper.serialize(model);
	}

	@Override
	public CaptainModel deserialize(final String body, final Class<CaptainModel> modelClass) {
		final JsonMapper jsonMapper = new JsonMapper();
		return jsonMapper.deserialize(body, modelClass);
	}
}

File: sample-data-repository/src/main/scala/nl/haploid/octowight/sample/data/CaptainModelSerializer.scala
package nl.haploid.octowight.sample.data

import nl.haploid.octowight.JsonMapper
import nl.haploid.octowight.registry.data.ModelSerializer
import org.springframework.stereotype.Component

@Component
class CaptainModelSerializer extends ModelSerializer[CaptainModel] {
  def serialize(model: CaptainModel) = {
    val jsonMapper = new JsonMapper
    jsonMapper.serialize(model)
  }

  def deserialize(body: String, modelClass: Class[CaptainModel]) = {
    val jsonMapper = new JsonMapper
    jsonMapper.deserialize(body, modelClass)
  }
}
